cmd/kubeadm/app/util/path.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"path/filepath"
+	"runtime"
+)
+
+// NormalizePath takes a slash-separated path (which would be a valid unix path) and ensures it is in the correct
+// format for the os in which it is currently running (currently used to convert unix path to windows path)
+// ex given "/etc/kubernetes/pki/ca.crt" on windows, it will return "C:\etc\kubernetes\pki\ca.crt"
+// on linux, the call is a no-op
+func NormalizePath(path string) string {
+	if runtime.GOOS == "windows" {
+		if filepath.VolumeName(path) == "" {
```

Comment:
same thing: maybe split it to path_windows.go and build only on windows ?

Reply:
@kad would it make more sense just to split up defaults.go for necessary OS specific defaults (at the moment just one path), and get rid of the NormalizePath() func?

Reply:
I don't know. probably @luxas has opinion about that ?